user/test/com/google/gwt/emultest/java8/lang/MathTest.java

```diff
@@ -294,6 +334,172 @@ public void testToIntExact() {
+    // Any finite number divided by infinity -> same number
+    assertEquals(5.0, Math.IEEEremainder(5.0, Double.POSITIVE_INFINITY), EPS);
+    assertTrue(Double.isNaN(Math.IEEEremainder(Double.NaN, 2.0)));
+    assertTrue(Double.isNaN(Math.IEEEremainder(5.0, Double.NaN)));
+    assertTrue(Double.isNaN(Math.IEEEremainder(Double.NaN, Double.NaN)));
+  }
+
+  public void testIEEEremainder() {
+    assertEquals(1.0, Math.IEEEremainder(hideFromCompiler(7.0), 3.0), EPS);
+    assertEquals(-1.0, Math.IEEEremainder(hideFromCompiler(8.0), 3.0), EPS);
+    assertEquals(0.0, Math.IEEEremainder(hideFromCompiler(6.0), 3.0), EPS);
+    assertEquals(0.0, Math.IEEEremainder(hideFromCompiler(9.0), 3.0), EPS);
+    assertEquals(-1.0, Math.IEEEremainder(hideFromCompiler(-7.0), 3.0), EPS);
+    assertEquals(1.0, Math.IEEEremainder(hideFromCompiler(7.0), -3.0), EPS);
+    assertEquals(-1.0, Math.IEEEremainder(hideFromCompiler(-7.0), -3.0), EPS);
+    assertEquals(0.5, Math.IEEEremainder(hideFromCompiler(2.5), 1.0), EPS);
+    assertEquals(0.5, Math.IEEEremainder(hideFromCompiler(2.5), 2.0), EPS);
+    assertEquals(0.2, Math.IEEEremainder(hideFromCompiler(5.2), 1.0), EPS);
+
+    assertEquals(0.0, Math.IEEEremainder(hideFromCompiler(4.5), 1.5), EPS);
+    assertEquals(1.5, Math.IEEEremainder(hideFromCompiler(7.5), 3.0), EPS);
+    assertEquals(-1.5, Math.IEEEremainder(hideFromCompiler(-7.5), -3.0), EPS);
+    assertEquals(1.5, Math.IEEEremainder(hideFromCompiler(7.5), -3.0), EPS);
+    assertEquals(-1.5, Math.IEEEremainder(hideFromCompiler(-7.5), 3.0), EPS);
+    // Remainder with 0 divisor is NaN
+    assertTrue(Double.isNaN(Math.IEEEremainder(hideFromCompiler(5.0), 0.0)));
+    // 0 divided by anything is 0
+    assertEquals(0.0, Math.IEEEremainder(hideFromCompiler(0.0), 2.0), EPS);
+    // Infinity cases produce NaN
+    assertTrue(Double.isNaN(Math.IEEEremainder(hideFromCompiler(Double.POSITIVE_INFINITY), 2.0)));
+    assertTrue(Double.isNaN(Math.IEEEremainder(hideFromCompiler(Double.NEGATIVE_INFINITY), 2.0)));
+    assertEquals(2, Math.IEEEremainder(hideFromCompiler(2), Double.POSITIVE_INFINITY), EPS);
+    assertEquals(2, Math.IEEEremainder(hideFromCompiler(2), Double.NEGATIVE_INFINITY), EPS);
+    // Any finite number divided by infinity -> same number
+    assertEquals(5.0, Math.IEEEremainder(hideFromCompiler(5.0), Double.POSITIVE_INFINITY), EPS);
+    assertTrue(Double.isNaN(Math.IEEEremainder(hideFromCompiler(Double.NaN), 2.0)));
+    assertTrue(Double.isNaN(Math.IEEEremainder(hideFromCompiler(5.0), Double.NaN)));
+    assertTrue(Double.isNaN(Math.IEEEremainder(hideFromCompiler(Double.NaN), Double.NaN)));
+  }
+
+  public void testHypot() {
+    assertEquals(5.0, Math.hypot(3.0, 4.0), EPS);
+    assertEquals(Double.POSITIVE_INFINITY, Math.hypot(1, Double.NEGATIVE_INFINITY));
+    assertEquals(Double.POSITIVE_INFINITY, Math.hypot(Double.POSITIVE_INFINITY, 1));
+    assertEquals(Double.POSITIVE_INFINITY, Math.hypot(Double.NaN, Double.NEGATIVE_INFINITY));
```

Comment:
warning, these are boxing to Double without the third arg (either 0 or EPS in other lines)